src/db.rs

```diff
@@ -626,6 +690,24 @@ impl Db {
         self.inner.delete_with_options(key, options).await;
     }
 
+    /// Write a batch of put/delete operations atomically to the database. All batch
+    /// operations are guaranteed to end up in the same WAL SST (or L0 memtable if
+    /// WAL is disabled).
+    pub async fn write_batch(&self, batch: WriteBatch) {
+        self.write_batch_with_options(batch, DEFAULT_WRITE_OPTIONS)
+            .await;
+    }
+
+    /// Write a batch of put/delete operations atomically to the database. All batch
+    /// operations are guaranteed to end up in the same WAL SST (or L0 memtable if
```

Comment:
curious why this is important? (ending up in the same WAL SST)

Reply:
in my understanding, making them in a same WAL SST may help make it easier to make the writes atomic.

object storage somehow makes this atomicity easier in my understanding, we can either put the entire file successfully or put nothing in the object store.

if we seperated the writes in multiple wal files, we need some metadata header which manages about how many keys are contained in this batch, and skip the inconsistent wals when it's written in half.

this comment may related with this:

https://github.com/slatedb/slatedb/pull/264#discussion_r1801171589

Reply:
Currently with wal-enabled a single WAL SST write is required to make the whole batch atomic. Once a WAL SST is written successfully, the data is part of the db. If the batch is split across WAL SSTs then the write could fail to write one of the WAL SSTs. This isn't strictly true of the L0 memtable write protocol - we could split the batch up over multiple memtables and atomically write them to the db as part of the same manifest update. Maybe it's better not to specify this here since it's an implementation detail.

Reply:
Removed confusing comment: 4997549